server/data/data-ldap/src/test/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepositoryTest.java

```diff
@@ -46,183 +52,620 @@
 
-            assertThat(usersLDAPRepository.supportVirtualHosting()).isTrue();
+            assertThat(usersRepository.test(JAMES_USER_MAIL, PASSWORD)).isTrue();
         }
 
         @Test
-        void supportVirtualHostingShouldReturnFalseWhenReportedInConfig() throws Exception {
-            HierarchicalConfiguration<ImmutableNode> configuration = ldapRepositoryConfiguration();
-            configuration.addProperty(ReadOnlyUsersLDAPRepository.SUPPORTS_VIRTUAL_HOSTING, "false");
+        void knownUserShouldNotBeAbleToLogInWhenPasswordIsNotCorrectWithVirtualHosting() throws Exception {
+            assertThat(usersRepository.test(JAMES_USER, BAD_PASSWORD)).isFalse();
+        }
 
-            ReadOnlyUsersLDAPRepository usersLDAPRepository = new ReadOnlyUsersLDAPRepository(domainList);
-            usersLDAPRepository.configure(configuration);
+        @Test
+        void unknownUserShouldNotBeAbleToLogInWithVirtualHosting() throws Exception {
+            assertThat(usersRepository.test(UNKNOWN, BAD_PASSWORD)).isFalse();
+        }
 
-            assertThat(usersLDAPRepository.supportVirtualHosting()).isFalse();
+        @Test
+        void unknownUserShouldNotBeAbleToLogInWhenPasswordIsCorrectWithVirtualHosting() throws Exception {
+            assertThat(usersRepository.test(UNKNOWN, PASSWORD)).isFalse();
         }
 
         @Test
-        void configureShouldThrowOnNonBooleanValueForSupportsVirtualHosting() {
-            HierarchicalConfiguration<ImmutableNode> configuration = ldapRepositoryConfiguration();
-            configuration.addProperty(ReadOnlyUsersLDAPRepository.SUPPORTS_VIRTUAL_HOSTING, "bad");
+        void specialCharacterInUserInputShouldBeSanitized() throws Exception {
+            Username patternMatchingMultipleUsers = Username.of("j*");
 
-            ReadOnlyUsersLDAPRepository usersLDAPRepository = new ReadOnlyUsersLDAPRepository(domainList);
+            assertThat(usersRepository.test(patternMatchingMultipleUsers, PASSWORD)).isFalse();
+        }
 
-            assertThatThrownBy(() -> usersLDAPRepository.configure(configuration))
-                .isInstanceOf(ConversionException.class);
+        @Test
+        void containsWithGetUserShouldBeTrueWithVirtualHosting() throws Exception {
+            assertThat(usersRepository.contains(usersRepository.getUser(JAMES_USER_MAIL.asMailAddress()))).isTrue();
+        }
+
+        @Disabled("JAMES-3088 isAdministrator is case sensitive")
+        @Override
+        @Test
+        public void isAdministratorShouldBeCaseInsentive(TestSystem testSystem) throws Exception {
+        }
+
+        @Disabled("JAMES-3088 This user-repository is read-only. Modifications are not permitted.")
+        @Override
+        @Test
+        public void isAdministratorShouldReturnFalseWhenNotConfigured(TestSystem testSystem) throws Exception {
+        }
+
+        @Disabled("JAMES-3088 This user-repository is read-only. Modifications are not permitted.")
```

Comment:
I'm not sure of e27cbf7794aa9336a7ed356b723655c2fd091bd5 commit, that's a looooot of disabled tasts. 

How many tests are actually played from Abstract Users repository contacts?

I suggest splitting the Abstract users repository contracts in 2: the ones that requires add/remove (and can't be played over LDAP) and the ones that only requires other methods (and can be played over LDAP).

Hence, in LDAP tests we should only implement the contracts we pass.

Thus see no

```
@Disabled("JAMES-3088 This user-repository is read-only. Modifications are not permitted.")
```